### PR TITLE
feat: use `@prismicio/client` v6 internally and update dependencies

### LINKED ISSUE #38: Upgrade vulnerable dependency
There is a severe vulnerability in one of the packages transitive dependencies. Output of `npm audit`:

```
node-fetch  <2.6.7
Severity: high
node-fetch is vulnerable to Exposure of Sensitive Information to an Unauthorized Actor - https://github.com/advisories/GHSA-r683-j2x4-v87g
No fix available
node_modules/@firebase/firestore/node_modules/node-fetch
node_modules/@firebase/functions/node_modules/node-fetch
node_modules/@firebase/storage/node_modules/node-fetch
node_modules/cross-fetch/node_modules/node-fetch
  @firebase/firestore  <=0.0.900-exp.f919db6a9 || 1.16.2-20206244562 - 1.16.2-canary.f1299756 || >=1.16.3-20206291717
  Depends on vulnerable versions of node-fetch
  node_modules/@firebase/firestore
    firebase  0.800.3 - 0.900.25 || 7.9.1-0 - 7.9.1-canary.0396117e || 7.17.1-20206244562 - 7.17.1-canary.f1299756 || >=7.17.2-20206291717
    Depends on vulnerable versions of @firebase/firestore
    Depends on vulnerable versions of @firebase/functions
    Depends on vulnerable versions of @firebase/storage
    node_modules/firebase
  @firebase/functions  <=0.0.900-exp.f919db6a9 || 0.4.51-202088235442 - 0.4.51-eap-auth-emulator.df41ee388 || >=0.5.0-20209118324
  Depends on vulnerable versions of node-fetch
  node_modules/@firebase/functions
  @firebase/storage  <=0.0.900-exp.f43d0c698 || 0.5.5-202151602035 - 0.5.5-canary.f6e1645ef || 0.5.6-20216122160 - 0.9.1
  Depends on vulnerable versions of node-fetch
  node_modules/@firebase/storage
  cross-fetch  <=3.1.4 || >=3.2.0-alpha.0
  Depends on vulnerable versions of node-fetch
  node_modules/cross-fetch
    prismic-javascript  1.5.0-beta.0 - 2.7.1
    Depends on vulnerable versions of cross-fetch
    node_modules/prismic-javascript
      apollo-link-prismic  *
      Depends on vulnerable versions of prismic-javascript
      node_modules/apollo-link-prismic
```

Since `prismic-javascript` is deprecated and is not expected to receive any updates or fixes according to your deprecation notice, it means that you have to **replace the `prismic-javascript` dependency with `@prismicio/client`**.

Thanks for your help!

### PR DESCRIPTION
**Pull request by angeloashmore**
## Types of changes

- [ ] Chore (a non-breaking change which is related to package maintenance)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR re-implements `apollo-link-prismic` using `@prismicio/client` v6's `graphqlFetch()` method  internally. This simplifies the Link greatly.

It also updates all dependencies to the most up-to-date versions.

Lastly, it updates the repository to use the latest Prismic TypeScript project standards.

Fixes: #38

## Checklist:

- [x] My change requires an update to the official documentation.
- [x] All [TSDoc](https://tsdoc.org) comments are up-to-date and new ones have been added where necessary.
- [x] All new and existing tests are passing.

👩‍🚀

### CODE REVIEW

**Comment by angeloashmore** [comment]
⚠️ **TODO**: Re-enable Size Limit in `.github/workflows/ci.yml` after merging into `master`.

File: src/createPrismicLink.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * Configuration for `createPrismicLink`.
+ */
+export type PrismicLinkConfig = Omit<
+	HttpOptions,
+	"fetch" | "useGETForQueries"
+> & {
+	/**
+	 * The name of the link's Prismic repository.
+	 */
+	repositoryName: string;
```
Comment (bencehornak) [inline src/createPrismicLink.ts]: `repositoryName` could be optional, because it is derived from `uri` if that is present

**Comment by bencehornak** [comment]
@angeloashmore it integrated well into our project.

Just one more thing: although the tsdoc comment mentions that the `fetch` parameter is needed on Node, I would consider highlighting this in the README too with a Node specific example:

```typescript
import { ApolloClient, InMemoryCache } from "@apollo/client";
import { createPrismicLink } from "apollo-link-prismic";
import fetch from "node-fetch";

const apolloClient = new ApolloClient({
	link: createPrismicLink({
		repositoryName: "YOUR_REPOSITORY_NAME",
		// Provide your access token if your repository is secured.
		accessToken: "YOUR_ACCESS_TOKEN",
		fetch: fetch,
	}),
	cache: new InMemoryCache(),
});
```

**Comment by angeloashmore** [comment]
@bencehornak Thanks for trying it out!

Good idea about documenting the need to pass a `fetch` implementation. The official documentation (https://prismic.io/docs/technologies/integrate-with-existing-js-project-graphql) will be updated soon to reflect the changes made here. You'll see the docs are outdated (for example, they don't use `@apollo/client`). I'll be sure to note that passing `fetch` is required in `fetch`-less environments.

Until those docs are updated, I'll update the README like you suggest.

File: src/createPrismicLink.ts (hunk 1 above)
Comment (angeloashmore) [reply src/createPrismicLink.ts]: Thanks for catching that. I've updated the type to support at least either `repositoryName` or `uri`.

**Comment by angeloashmore** [comment]
@bencehornak `apollo-link-prismic@1.1.0-alpha.2` was just published with the type fix. Could you confirm that this matches what you expect?

If it does, I'll move forward with publishing this as latest. Thanks!

**Comment by bencehornak** [comment]
@angeloashmore now the typing works great as well!

One final thought: since this PR brings in breaking changes (e.g. now the fetch argument is required in some environments), wouldn't it be better to bump the major version? ([semver \#8](https://semver.org/#spec-item-8))

**Comment by angeloashmore** [comment]
@bencehornak Correct me if your experience is different, but the current version also requires `fetch` to be provided by the environment or the `fetch` option.

It isn't documented in the current README or in the code. It uses Apollo's HttpLink internally (as does the updated version), which needs a global or provided `fetch` function ([see this section of the Apollo docs](https://github.com/apollographql/apollo-client/blob/b214dd1d31cdfe6b5477bb5887a7d9615a302b27/docs/source/networking/advanced-http-networking.md#providing-a-fetch-replacement-for-certain-environments)).

So the API _should_ be the same, but it is now explicitly documented in the form of TSDocs and the README (and eventually in the official Prismic docs).

**Comment by bencehornak** [comment]
@angeloashmore you must be right.

(The reason, why I was confused, is that I actually had to add the `fetch` property now. It worked so far without it, because of the presence of a polyfill, provided by some of our dependencies. `@prismicio/client` for some reason didn't pick up the polyfill, so I had to provide `fetch` explicitly. I don't think this is a bug with `@prismicio/client`, it's rather a problem of the dependency, which loaded the polyfill in a non-standard way.)

Anyway, I think this sounds like an edge-case, most users shouldn't experience a breaking change here, so the minor version bump should be fine.

**Comment by angeloashmore** [comment]
Thanks for testing it out @bencehornak.

This PR is published in `v1.1.0` 🎉

**Pull request merged**